rq/worker.py

```diff
@@ -572,6 +572,8 @@ def work(self, burst=False, logging_level="INFO", date_format=DEFAULT_LOGGING_DA
                         'Worker %s: found an unhandled exception, quitting...',
                         self.key, exc_info=True
                     )
+                    if self.is_horse:
+                        os._exit(1)
```

Comment:
We should move `os._exit(1)` to a `try/except` block around `perform_job()` in line 789 so we catch this error much earlier.